plugins/sdexFeed.go

```diff
@@ -0,0 +1,73 @@
+package plugins
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/interstellar/kelp/support/utils"
+	"github.com/stellar/go/build"
+	"github.com/stellar/go/clients/horizon"
+)
+
+type sdexFeed struct {
+	sdex       *SDEX
+	assetBase  *horizon.Asset
+	assetQuote *horizon.Asset
+}
+
+// newSDEXFeed creates a price feed from buysell's url fields
+func newSDEXFeed(sdex *SDEX, url string) (*sdexFeed, error) {
+	s := new(sdexFeed)
+	s.sdex = sdex
+	urlParts := strings.Split(url, "/")
+
+	baseURL := strings.Split(urlParts[0], ":")
+	baseCode := baseURL[0]
+	baseIssuer := baseURL[1]
+	baseConvert, e := parseAsset(baseCode, baseIssuer)
+	if e != nil {
+		return nil, fmt.Errorf("unable to convert base asset url to sdex asset")
+	}
+	s.assetBase = baseConvert
+
+	quoteURL := strings.Split(urlParts[1], ":")
+	quoteCode := quoteURL[0]
+	quoteIssuer := quoteURL[1]
+	quoteConvert, e := parseAsset(quoteCode, quoteIssuer)
+	if e != nil {
+		return nil, fmt.Errorf("unable to convert quote asset url to sdex asset")
+	}
+	s.assetQuote = quoteConvert
+	return s, nil
+}
+
+func (s *sdexFeed) GetPrice() (float64, error) {
+	orderBook, e := utils.GetOrderBook(s.sdex.API, s.assetBase, s.assetQuote)
+	if e != nil {
+		return 0, fmt.Errorf("unable to get sdex price: %s", e)
+	}
+	bids := orderBook.Bids
+	topBidPrice := utils.PriceAsFloat(bids[0].Price)
+	asks := orderBook.Asks
+	lowAskPrice := utils.PriceAsFloat(asks[0].Price)
+	centerPrice := (topBidPrice + lowAskPrice) / 2
+	return centerPrice, nil
+}
+
+func parseAsset(code string, issuer string) (*horizon.Asset, error) {
```

Comment:
ok, makes sense to generalize -- then the logic in trader/config.go#Init() should wrap the returned error from parseAsset() with the related "A" or "B" error message instead of just returning the error (trader/config.go line 81 and line 87)